### DISCUSSION TITLE
After compile all code in Windows 7 ，rustdesk can cannot run，show dll and exception

### DISCUSSION THREAD

**Discussion by longzhou3** [rustdesk/rustdesk#549] [General]
- environment：win7
1. cargo build
2. cargo run ： error: process didn't exit successfully: `target\debug\rustdesk.exe` (exit code: 0xc0000135, STATUS_DLL_NOT_FOUND)
double click show dll lost api-ms-win-shcore-scaling-l1-1-1.dll
3. add api-ms-win-shcore-scaling-l1-1-1.dll
4. error: process didn't exit successfully: `target\debug\rustdesk.exe` (exit code: 0xc000007b)

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/blob/master/build.py
But I compiled on win7 with Sciter static lib, not sure about dyn lib. If you wanna use Sciter static lib, you need buy from Sciter author.

**Reply by longzhou3** [reply]
thanks，i try to package using build.py !

**Comment by yukitteru** [comment]
> * environment：win7
> 
> 1. cargo build
> 2. cargo run ： error: process didn't exit successfully: `target\debug\rustdesk.exe` (exit code: 0xc0000135, STATUS_DLL_NOT_FOUND)
>    double click show dll lost api-ms-win-shcore-scaling-l1-1-1.dll
> 3. add api-ms-win-shcore-scaling-l1-1-1.dll
> 4. error: process didn't exit successfully: `target\debug\rustdesk.exe` (exit code: 0xc000007b)


Win7 is missing api-ms-win-shcore-scaling-l1-1-1.dll.
https://docs.microsoft.com/en-us/windows/win32/api/shellscalingapi/nf-shellscalingapi-setprocessdpiawareness

 ```
#[cfg(all(windows, not(feature = "inline")))]
    unsafe {
        winapi::um::shellscalingapi::SetProcessDpiAwareness(2);
    }
```
Fix it in [ui.rs](https://github.com/rustdesk/rustdesk/blob/master/src/ui.rs)

**Reply by rustdesk** [reply]
Good job, can you create a PR?

**Reply by rustdesk** [reply]
for example, check if win7 in build.rs, then disable winapi::um::shellscalingapi::SetProcessDpiAwareness(2)

**Reply by yukitteru** [reply]
> например, проверьте, есть ли win7 в build.rs, затем отключите winapi::um::shellscalingapi::SetProcessDpiAwareness(2)

a lil bit later
